### PR TITLE
Clown anomalous crystals turn the dead into living clowns, rather than anyone who speaks in proximity

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Rather than turning people into clowns if they speak within proximity of the crystal, the clown anomalous crystal instead transforms the dead into clowns, similar to the dark revival crystal.

This also adds logging for crystal activation.

## Why It's Good For The Game

We've had a significant number of admin issues with this crystal being abused to both grief and killbait people by turning them into clowns as of late. The crystal literally only exists to do this basically. And it is also grossly unlogged despite the fact that many of the effects could be cause for investigation.

Rather remove this crystal, I'm making it useful...with a catch.

## Changelog
:cl:
qol: Makes the clown anomalous crystal actually useful for once rather than an open griefing tool. It now revives the dead...as clowns!
admin: Adds logging for the anomalous crystal activation, including fingerprints.
/:cl:

### CODE REVIEW

**Comment by KingkumaArt** [comment]
this is like the ss13 equivalent of "sometimes, dead is better"

**Comment by Jackal-boop** [comment]
significantly less funny

**Comment by Cheshify** [comment]
Sick of this thing being a ticket & ahelp generator

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 1]
```diff
@@ -328,22 +329,28 @@
 /obj/machinery/anomalous_crystal/honk/ActivationReaction(mob/user)
 	. = ..()
 	if(!.)
-		return FALSE
+		return
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:332]: why are we removing the return values?

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 2]
```diff
@@ -328,22 +329,28 @@
...
+			continue
 
-	if(!ishuman(user))
-		return FALSE
-	if(!(user in viewers(src)))
-		return FALSE
-	var/clown_ref = REF(user)
-	if(clown_ref in clowned_mob_refs)
-		return FALSE
+		if(ishuman(thing))
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:338]: invert this for an early continue

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 3]
```diff
@@ -317,9 +318,9 @@
 	ActivationReaction(null, ACTIVATE_BOMB)
 	return TRUE
 
-/obj/machinery/anomalous_crystal/honk //Strips and equips you as a clown. I apologize for nothing
-	observer_desc = "This crystal strips and equips its targets as clowns."
-	possible_methods = list(ACTIVATE_MOB_BUMP, ACTIVATE_SPEECH)
+/obj/machinery/anomalous_crystal/honk //Revives the dead, but strips and equips them as a clown
+	observer_desc = "This crystal revives targets around it as clowns Oh, that's horrible...."
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:322]:
```suggestion
	observer_desc = "This crystal revives targets around it as clowns."
```

i don't really know what you meant here

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by Stoopler** [comment]
I think this is going to wind up a lot like people not wanting to be put in a MMI and borged if anyone can revive them as a clown.

Instead of completely ripping apart the gimmick of the item, could we make it give feedback to the player/nearby players that are activating it that something feels "off" and that if they continue talking they will suffer consequences?  Give it like 3 triggers within a 10 second span or something like that.

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm (hunk 3 above)
Comment (necromanceranne) [reply code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:322]: That becoming a clown is a fate worse than death, I guess? I just missed a full stop.

**Comment by necromanceranne** [comment]
I literally don't give a shit to retain the functionality it has whatsoever, while it is still a policy problem. I'm getting tired of it being a constant headache for myself and my fellow admins to have to resolve.

Your first argument is nonsense. Revival is trivial, and as a result borging is regarded as a punishment/last resort at best and we treat it as such in the rules. The benefit to mass revivals like these two crystals is the extreme speed at which it brings people back into the round.

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm (hunk 3 above)
Comment (san7890) [reply code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:322]: fine then, just lacked grammatical sense

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 4]
```diff
@@ -329,21 +330,30 @@
...
+		new_clown.revive(ADMIN_HEAL_ALL, force_grab_ghost = TRUE)
+
+		var/clown_ref = REF(new_clown)
+		if(clown_ref in clowned_mob_refs) //one clowning per person
+			continue
+			
+		for(var/obj/item/to_strip in new_clown.get_equipped_items())
+			new_clown.dropItemToGround(to_strip)
+		new_clown.dress_up_as_job(SSjob.GetJobType(/datum/job/clown))
+		clowned_mob_refs += clown_ref
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:356]:
```suggestion
		clowned_mob_refs += clown_ref
	
	return TRUE
```

File: code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm [hunk 5]
```diff
@@ -402,21 +412,26 @@
...
+		var/mob/living/carbon/human/to_revive = thing
+		
+		if(to_revive.stat != DEAD)
+			continue
+		
+		to_revive.set_species(/datum/species/shadow, TRUE)
+		to_revive.revive(ADMIN_HEAL_ALL, force_grab_ghost = TRUE)
+		//Free revives, but significantly limits your options for reviving except via the crystal
+		//except JK who cares about BADDNA anymore. this even heals suicides.
+		ADD_TRAIT(to_revive, TRAIT_BADDNA, MAGIC_TRAIT)
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm:434]:
```suggestion
		ADD_TRAIT(to_revive, TRAIT_BADDNA, MAGIC_TRAIT)
		
	return TRUE
```

**Review by san7890 — CHANGES_REQUESTED** [review]
let's get it to be consistent (one of them was already returning true anyways)
image

**Comment by zoomachina** [comment]
this isn't a "quality of life" change

**Comment by necromanceranne** [comment]
it is for me

**Comment by san7890** [comment]
that is a good point i forgot to look at the labels. it doesn't matter though because it's getting merged because i can see how it's way overpowered anyways

**Pull request merged**